File: D1P1.java
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Scanner;
import java.util.List;
import java.io.File;
import java.io.FileNotFoundException;

public class D1P1{
  public static void main(String[] args){
    Scanner in = new Scanner(System.in);
    System.out.print("Please enter file name: ");
    String fileName = in.nextLine();

    List<Integer> left = new ArrayList<>();
    List<Integer> right = new ArrayList<>();

    try{
      File file = new File(fileName);
      Scanner fileReader = new Scanner(file);

      while (fileReader.hasNextLine()){
        left.add(fileReader.nextInt());
        right.add(fileReader.nextInt());
      }

      left.sort(null);
      right.sort(null);

      int sum = 0;
      for (int i = 0; i < left.size(); i++){
        int diff = Math.abs(left.get(i) - right.get(i));
        sum += diff;
      }

      System.out.println("Total distance: " + sum);
    }
    catch (FileNotFoundException e){
      System.out.println("File not found!");
    }

  }
}

File: D1P1.py
fileName = input('Enter file name: ')

# https://www.w3schools.com/python/python_try_except.asp
# https://stackoverflow.com/questions/5627425/what-is-a-good-way-to-handle-exceptions-when-trying-to-read-a-file-in-python

try:
  with open(fileName, 'r') as f:
    left = []
    right = []

    for line in f:
      values = line.split()
      left.append(int(values[0]))
      right.append(int(values[1]))

    left.sort()
    right.sort()

    total_distance = 0
    for i in range(len(left)):
      total_distance += abs(left[i] - right[i])
    
    print(f'Total distance: {total_distance}')
except FileNotFoundError:
  print(f'{fileName} not found!')